src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -136,7 +136,7 @@ public string CreateToken(JObject payload, SigningCredentials signingCredentials
         /// <param name="signingCredentials">Defines the security key and algorithm that will be used to sign the JWT.</param>
         /// <param name="encryptingCredentials">Defines the security key and algorithm that will be used to encrypt the JWT.</param>
         /// <returns>A JWT in compact serialization format.</returns>
-        private string CreateJsonWebToken(JObject payload, SigningCredentials signingCredentials, EncryptingCredentials encryptingCredentials)
+        public string CreateToken(JObject payload, SigningCredentials signingCredentials, EncryptingCredentials encryptingCredentials)
         {
             if (payload == null)
```

Comment:
null check on payload, signingCredentials, encryptingCredentials.
then call private CreateToken.
Move check for cached header in private CreateToken, no need to pass as parameter.

Reply:
Done.